Sources/SwiftParser/Statements.swift

```diff
@@ -255,7 +254,14 @@ extension Parser {
       let pattern = self.parseMatchingPattern(context: .matching)
       kind = .pattern(caseKeyword, pattern)
     } else {
-      let letOrVar = self.consumeAnyToken()
+      let letOrVar: RawTokenSyntax
+
+      if self.at(.identifier) {
+        letOrVar = RawTokenSyntax(missing: lastBindingKind!.tokenKind, text: lastBindingKind!.tokenText, arena: self.arena)
```

Comment:
Could you either add a comment describing why `lastBindingKind` cannot be `nil` here or (and I’d probably prefer that) add `let lastBindingKind = lastBindingKind` to the `if` condition?

---

I think you don’t need to specify `text`

---

Could you add a comment that we’re in the case where the the user forgot to write `let` or `var`?